nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -385,10 +363,30 @@ Costmap2DROS::mapUpdateLoop(double frequency)
     return;
   }
 
-  RCLCPP_DEBUG(get_logger(), "Entering loop");
+  // First, make sure that the transform between the robot base frame
+  // and the global frame is available
 
+  std::string tf_error;
   rclcpp::Rate r(frequency);    // 200ms by default
 
+  RCLCPP_INFO(get_logger(), "Checking transform");
+  while (rclcpp::ok() &&
+    !tf_buffer_->canTransform(
+      global_frame_, robot_base_frame_, tf2::TimePointZero, &tf_error))
+  {
+    RCLCPP_INFO(
+      get_logger(),
+      "Timed out waiting for transform from %s to %s to become available, tf error: %s",
+      robot_base_frame_.c_str(), global_frame_.c_str(), tf_error.c_str());
+
+    // The error string will accumulate and errors will typically be the same, so the last
+    // will do for the warning above. Reset the string here to avoid accumulation
+    tf_error.clear();
+    r.sleep();
```

Comment:
This rate should be separate of the costmap update rate. This will spam the console a bunch. I actually had to do a bunuch of TF stuff to make that work out so that it was minimized :laughing:, that should be hardcoded at 2hz. I dont want anyone changing that by accident.

Reply:
Ah I was wondering why that particular rate was hardcoded